### ISSUE TITLE
Message is not routed to all session in different nodes.

### ISSUE THREAD

**Issue by iPPLE** [processone/ejabberd#3883]
I am running a cluster of 2 nodes, which having these following sessions in `sm` table

| username | node |  resource | priority |
|-----------|------|----------|---------|
| alice | ejabberd@node1.com | android | 0 |
| alice | ejabberd@node2.com | android | 0 |
| bob | ejabberd@node2.com | android | 0 |

When `bob` send message to `alice` only `alice` session which is in the same node as bob (`node2`) can receive the message.

* `ejabberd-21.01.7`
* Ejabberd clustering is up properly means either node is down the service still continue to work.

Please help to advice.
Thanks.

**Comment by badlop** [comment]
Are you sending the exact same stanza when it works and later when it fails?

This stanza will be delivered only to one client, the one connected in node1:
```xml
<message id='27:193807'
	type='chat'
	to='alice@localhost/alice_001'>
  <body>hi first node</body>
</message>
```

This stanza will be delivered to all the connected clients, the one in node1 and the one in node2:
```xml
<message id='27:193807'
	type='chat'
	to='alice@localhost'>
  <body>hi all</body>
</message>
```

**Comment by iPPLE** [comment]
> Are you sending the exact same stanza when it works and later when it fails?
> 
> This stanza will be delivered only to one client, the one connected in node1:
> ```xml
> <message id='27:193807'
> 	type='chat'
> 	to='alice@localhost/alice_001'>
>   <body>hi first node</body>
> </message>
> ```
> 
> This stanza will be delivered to all the connected clients, the one in node1 and the one in node2:
> ```xml
> <message id='27:193807'
> 	type='chat'
> 	to='alice@localhost'>
>   <body>hi all</body>
> </message>
> ```
> 
> 

Yes I sent the same message as always. I think server side manage this resources suffix..
